common/src/main/java/tds/common/web/interceptors/RestTemplateLoggingInterceptor.java

```diff
@@ -0,0 +1,88 @@
+package tds.common.web.interceptors;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Charsets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+ * An interceptor for logging REST requests and responses
+ */
+public class RestTemplateLoggingInterceptor implements ClientHttpRequestInterceptor {
+    private static final Logger log = LoggerFactory.getLogger(RestTemplateLoggingInterceptor.class);
+    private static final String CONTENT_TYPE_SUBTYPE_JSON = "json";
+    private ObjectMapper objectMapper;
+
+    public RestTemplateLoggingInterceptor(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public ClientHttpResponse intercept(final HttpRequest request, final byte[] body, final ClientHttpRequestExecution clientHttpRequestExecution) throws IOException {
+        logRequest(request, body);
+        ClientHttpResponse response = clientHttpRequestExecution.execute(request, body);
+        logResponse(response);
+        return response;
+    }
+
+    private void logRequest(final HttpRequest request, final byte[] body) {
+        final HttpHeaders headers = request.getHeaders();
+        String bodyString = new String(body, Charsets.UTF_8);
+
+        if (!bodyString.isEmpty() && headers.getAccept().contains(MediaType.APPLICATION_JSON)) {
```

Comment:
For the request, shouldn't you be checking the Content-Type rather than the Accept type?